### PR TITLE
Support multiple GitHub SSH deploy keys

### PR DESCRIPTION
**Pull request by eronnen**
This PR's goal is to solve [#340](https://github.com/game-ci/unity-builder/issues/340), hope it can be a good enough of a solution :)

#### Background 

The [webfactory/ssh-agent](https://github.com/webfactory/ssh-agent) action handles the issue of authenticating with multiple SSH deploy keys to Github by adding the multiple public keys to the `.ssh` folder with custom file names, and then using them accordingly by adding `url.*.insteadOf` git configs.

The problem is that these git configs as well as the SSH public keys are not uploaded to the container, so this mirroring doesn't work in the build.

#### Changes

- add `sshPublicKeysDirectoryPath` input, which is a custom `.ssh` that should contain the public keys and `config` file.
- pass a `GIT_CONFIG_EXTENSIONS` environment variable to the container, which is a list of `key=values` configs that can be retrieved from `git config --list` that the git inside the container will use.

#### Usage

With these changes, it's possible to fix the multiple SSH keys issue by using the following steps:

```yaml
- uses: webfactory/ssh-agent@v0.8.0
  with:
  ssh-private-key: |
    ${{ secrets.FIRST_KEY }}
    ${{ secrets.NEXT_KEY }}
    ${{ secrets.ANOTHER_KEY }}

- name: Prepare SSH config for unity builder
  run: |
    mkdir $HOME/.ssh_docker
    cp $HOME/.ssh/config $HOME/.ssh_docker/
    cp $HOME/.ssh/key* $HOME/.ssh_docker/
    cp $HOME/.ssh/known_hosts $HOME/.ssh_docker/
    sed -i 's/\/home\/runner/\/root/g' $HOME/.ssh_docker/config
    sudo chown -R root:root $HOME/.ssh_docker
    GIT_CONFIG_EXTENSIONS=$(git config --list | grep '^url\.')
    {
      echo 'GIT_CONFIG_EXTENSIONS<<EOF'
      echo "$GIT_CONFIG_EXTENSIONS"
      echo EOF
    } >> "$GITHUB_ENV"
  shell: bash

- uses: game-ci/unity-builder@v3
  ...
  with:
    sshAgent: ${{ env.SSH_AUTH_SOCK }}
    sshPublicKeysDirectoryPath: ${{ env.HOME }}/.ssh_docker
```

#### Checklist

- [x] Read the contribution [guide](https://github.com/game-ci/unity-builder/blob/main/CONTRIBUTING.md) and accept the
      [code](https://github.com/game-ci/unity-builder/blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Docs (If new inputs or outputs have been added or changes to behavior that should be documented. Please make
            a PR in the [documentation repo](https://github.com/game-ci/documentation))
- [ ] Readme (updated or not needed)
- [ ] Tests (added, updated or not needed)

### CODE REVIEW

File: src/model/docker.ts [hunk 1]
```diff
@@ -89,6 +112,13 @@ class Docker {
             --volume "${actionFolder}/default-build-script":"c:/UnityBuilderAction" \
             --volume "${actionFolder}/platforms/windows":"c:/steps" \
             --volume "${actionFolder}/BlankProject":"c:/BlankProject" \
+            ${sshAgent ? `--volume ${sshAgent}:c:/ssh-agent` : ''} \
+                ${
+                  sshAgent && !sshPublicKeysDirectoryPath
+                    ? `--volume c:/Users/Administrator/.ssh/known_hosts:c:/root/.ssh/known_hosts`
```
Comment (webbertakken) [inline src/model/docker.ts:118]: Could you explain where `c:/root` comes from? Have you tested that this works?

**Review by webbertakken — APPROVED** [review]
The approach looks elegant enough if you ask me.

What do you think @GabLeRoux @davidmfinol @AndrewKahr?

File: src/model/docker.ts (hunk 1 above)
Comment (eronnen) [reply src/model/docker.ts:118]: I tested only for linux runner, tried to port to windows as best as I can...
but looking at it again seems that `sshAgent` isn't used in Windows anyway so probably I should remove these windows changes enitrely

Comment (webbertakken) [reply src/model/docker.ts]: Yea better to have working software with missing features than the added complexity of a non working piece of which nobody knows how much of it was intended to work.

Thanks.

**Comment by davidmfinol** [comment]
Looks good to me.
Just need to raise a PR to update the docs page as well: https://github.com/game-ci/documentation/blob/main/docs/03-github/04-builder.mdx

**Comment by eronnen** [comment]
Thank you for the response!

added complementary documentation and `unity-test-runner` PRs:

* https://github.com/game-ci/unity-test-runner/pull/240
* https://github.com/game-ci/documentation/pull/416

**Pull request merged**